src/Subscriptions/Repositories/SubscriptionNotesRepository.php

```diff
@@ -0,0 +1,261 @@
+<?php
+
+namespace Give\Subscriptions\Repositories;
+
+use Give\Framework\Database\DB;
+use Give\Framework\Exceptions\Primitives\Exception;
+use Give\Framework\Exceptions\Primitives\InvalidArgumentException;
+use Give\Framework\Models\ModelQueryBuilder;
+use Give\Framework\Support\Facades\DateTime\Temporal;
+use Give\Helpers\Hooks;
+use Give\Log\Log;
+use Give\Subscriptions\Models\SubscriptionNote;
+use Give\Subscriptions\ValueObjects\SubscriptionNoteMetaKeys;
+use Give\Subscriptions\ValueObjects\SubscriptionNoteType;
+
+/**
+ * NOTE: This repository is still using the old comments table.
+ * In the future, we will migrate to the new comments table.
+ *
+ * @unreleased
+ */
+class SubscriptionNotesRepository
+{
+    /**
+     * @unreleased
+     *
+     * @var string[]
+     */
+    private $requiredSubscriptionProperties = [
+        'subscriptionId',
+        'content',
+    ];
+
+    /**
+     * @unreleased
+     */
+    private const COMMENT_TYPE = 'give_sub_note';
+
+    /**
+     * @unreleased
+     *
+     * @return SubscriptionNote|null
+     */
+    public function getById(int $noteId)
```

Comment:
Let's set the return type to `?SubscriptionNote` and remove the return tag from the doc block.